apps/emqx_authn/src/emqx_authn_utils.erl

```diff
@@ -230,15 +217,15 @@ without_password(Credential, [Name | Rest]) ->
             without_password(Credential, Rest)
     end.
 
-handle_var({var, Name}, undefined) ->
-    error({cannot_get_variable, Name});
+handle_var({var, _Name}, undefined) ->
+    <<>>;
```

Comment:
Might be better to return some impossible-to-match value here?  Otherwise, some predicates might be true accidentally?  For example: `${unknown1} =  ${unknown2}` would always be true, but that might be undesirable. :thinking:

Reply:
We should either return a binary, or raise a error here.
Raising a error leads to problems, because there may be real cases when the variable is sometimes defined and sometimes not, and in the same time there is a single template.
Any "real" values can clash with other user values.

So we chose to use empty value so that we naver clash with meaningful values.

Reply:
I think we need to mention it in the docs if we determine that it is replaced with empty string :thinking:

Reply:
@HJianBo added warnings in the docs: https://github.com/emqx/emqx-docs/pull/1369